ALMAcceleratorSampleSolution/Plugins/alm-plugins/PluginBase.cs

```diff
@@ -0,0 +1,265 @@
+//-----------------------------------------------------------------------
+// <copyright file="PluginBase.cs" company="Microsoft">
+// Copyright (c) 2022 All Rights Reserved
+// </copyright>
+// <summary>
+// Auto generated plugin base class
+// </summary>
+//-----------------------------------------------------------------------
+namespace Alm.Plugins
+{
+    using System;
+    using System.Runtime.CompilerServices;
+    using System.ServiceModel;
+    using Microsoft.Xrm.Sdk;
+    using Microsoft.Xrm.Sdk.Extensions;
+    using Microsoft.Xrm.Sdk.PluginTelemetry;
+
+    /// <summary>
+    /// Base class for all plug-in classes.
+    /// </summary>
+    public abstract class PluginBase : IPlugin
+    {
+        protected string PluginClassName { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PluginBase"/> class.
+        /// </summary>
+        /// <param name="pluginClassName">The <see cref=" cred="Type"/> of the plugin class.</param>
+        internal PluginBase(Type pluginClassName)
+        {
+            PluginClassName = pluginClassName.ToString();
+        }
+
+        /// <summary>
+        /// Main entry point for he business logic that the plug-in is to execute.
+        /// </summary>
+        /// <param name="serviceProvider">The service provider.</param>
+        /// <remarks>
+        /// </remarks>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Globalization", "CA1303:Do not pass literals as localized parameters", Justification = "Execute")]
+        public void Execute(IServiceProvider serviceProvider)
+        {
+            if (serviceProvider == null)
+            {
+                throw new InvalidPluginExecutionException(nameof(serviceProvider));
+            }
+
+            // Construct the local plug-in context.
+            var localPluginContext = new LocalPluginContext(serviceProvider);
+
+            localPluginContext.Trace($"Entered {PluginClassName}.Execute() " +
+                $"Correlation Id: {localPluginContext.PluginExecutionContext.CorrelationId}, " +
+                $"Initiating User: {localPluginContext.PluginExecutionContext.InitiatingUserId}");
+
+            try
+            {
+                // Invoke the custom implementation
+                ExecuteCdsPlugin(localPluginContext);
```

Comment:
Should this be Dataverse rather than CDS?

Reply:
@filcole PluginBase was auto generated file using pac. I agreed that name should be ExecuteDataversePlugin. Renamed the function.

Reply:
Thanks.  I logged that pac generates it incorrectly here: https://github.com/microsoft/powerplatform-vscode/issues/268